src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index > -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] - The `die` method uses `indexOf` and `splice` within a nested loop, which can be inefficient. Consider using the `filter` method to remove dead animals from the `Animal.alive` array. For example, you could replace the `die` method with a `filter` method that filters out animals with health less than or equal to 0.